src/include/stir/ProjDataInfoGeneric.inl

```diff
@@ -0,0 +1,273 @@
+/*
+Copyright 2017 ETH Zurich, Institute of Particle Physics and Astrophysics
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*!
+
+  \file
+  \ingroup projdata
+
+  \brief Implementation of inline functions of class stir::ProjDataInfoGeneric
+	
+  \author Parisa Khateri
+  \author Michael Roethlisberger
+*/
+
+// for sqrt
+#include <math.h>
+#include "stir/Bin.h"
+#include "stir/Succeeded.h"
+#include "stir/LORCoordinates.h"
+#include "stir/is_null_ptr.h"
+#include <algorithm>
+#include "stir/CartesianCoordinate3D.h"
+
+START_NAMESPACE_STIR
+
+//! find phi from correspoding lor
+float
+ProjDataInfoGeneric::get_phi(const Bin& bin)const
+{
+	LORInAxialAndNoArcCorrSinogramCoordinates<float> lor;
+	get_LOR(lor, bin);
+	if (bin.view_num()==0 && lor.phi()>0.1)
+		return lor.phi()-_PI;
```

Comment:
this is a very strange line. where does `0.1` come from?

Reply:
The output of ` lor.phi()` for `view=0` is either close to zero or is close to PI. I put `lor.phi()>0.1` instead of `lor.phi()>0` in the condition to avoid values which are close to zero (but not zero).

Reply:
we've put in changes in `master` such that `lor.phi()` no longer starts from zero. We'll have to merge those into the block code. So this condition will be unexpected then.

Also, 0.1 is not so small in radians.

Reply:
The changes in the `LORCoordinates` class has been made three years ago. I have updated this branch with more recent versions of the master branch (few months ago).  I get those outputs now when I call `lor.phi()` for `view=0`.

Yes, 0.1 is not too small but small enough not to be close to PI, and large enough not to be 0.

Reply:
Yes I agree. We test lor.phi() as well as the other coordinates. If there is a swap I suspect the test would fail?

Reply:
I guess the test should fail, but it will probably only start to fail when we introduce rotations.

Reply:
When I run the test with the current parisa:master get_s() gives opposite sign results (for test_projdata_info). The test_blocks_on_cylindrical_projectors crashes with an instance of 'H5::PropListException' which I have no idea what it is, but it happens with HDF5 ON or OFF

Reply:
comment in the right place?

question is if `test_projdata_info` worked at some point on your machine, and if so, what point that was :-)

`H5` stuff is more than likely a conflict between HDF5 libraries (between compilations and running). if it happens with HDF5 OFF, it means you haven't rebuilt STIR. (Re-setting it at the SuperBuild level  might not reset it at STIR level). In any case, `test_blocks_on_cylindrical_projectors` works on GHA (and probably Appveyor)

Reply:
YEs was not sure where to put the comment. Anyway,get_s() tests were working before I pulled the new version of parisa:master. Now the difference appear when view is zero (phi is actually the same) and the main difference is an opposite sign but occasionally some rounding difference too